assignments/array-methods.js

```diff
@@ -56,28 +57,52 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach(el => fullName.push(`${el.first_name} ${el.last_name}`));
+
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+allCaps = runners.map(runner => runner.first_name.toUpperCase() );
+console.log(allCaps);
+
+
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+largeShirts = runners.filter(runner => runner.shirt_size === 'L');
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+ticketPriceTotal = runners.reduce( (acc, cur) => acc + cur.donation, 0 );
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1 - Get list of runners that have XL, 2XL, or 3XL  shirt size
+let xLargeDonorList = [];
+xLargeDonorList = runners.filter( runner => runner.shirt_size.includes('XL'));
+console.log(xLargeDonorList);
+
+// Problem 2 - Instead of using list from Prob2, get total amount of donations of runners that have XL, 2XL, or 3XL  shirt size
+let largeDonorDonationTotal = [];
+largeDonorDonationTotal = runners.filter( runner => runner.shirt_size.includes('XL'))
+                                 .reduce( (acc, curr) => acc + curr.donation, 0);
+console.log(largeDonorDonationTotal);
+
+// Problem 3 - find max email length
+let max_emailLength = [];
+max_emailLength = runners.map(runner => runner.email.length);
```

Comment:
this is fun. I have an extra challenge for you. Try to find the max length using the reduce method.